modules/ecr/main.tf

```diff
@@ -12,25 +12,42 @@ resource "aws_ecr_repository" "ecr_repo" {
   }
 }
 
+# Create the list of account ids, works for both the parameters => ID and IDs.
+locals {
+  aws_principals_account_id_list = length(var.access_for_account_id) > 0 ? [
```

Comment:
A more clean approach is to just work here with the account_ids and do the arn formatting in line 44 one time.

Reply:
The clean approach has been performed.